.github/actions/tp-e2e-tests/entrypoint.sh

```diff
@@ -0,0 +1,211 @@
+		color_and_prefix "$gray_bg" "building Traffic Vault";
+	echo 'Starting Traffic Vault...';
+	docker run \
+		--detach \
+		--env-file="${ciab_dir}/variables.env" \
+		--hostname="${trafficvault}.infra.ciab.test" \
+		--name="$trafficvault" \
+		--publish=8087:8087 \
+		--rm \
+		"$trafficvault" \
+		/usr/lib/riak/riak-cluster.sh;
+}
+start_traffic_vault &
+
+sudo apt-get install -y --no-install-recommends gettext \
+	ruby ruby-dev libc-dev curl \
+	chromium-chromedriver postgresql-client \
+	gcc musl-dev
+
+sudo gem update --system && sudo gem install sass compass
+sudo npm i -g protractor@^7.0.0 forever bower grunt selenium-webdriver
+
+CONTAINER=$(docker ps | grep "selenium/node-chrome" | awk '{print $1}')
+CHROME_VER=$(docker exec "$CONTAINER" google-chrome --version | sed -E 's/.* ([0-9.]+).*/\1/')
+sudo webdriver-manager update --gecko false --standalone false --versions.chrome "LATEST_RELEASE_$CHROME_VER"
+
+GOROOT=/usr/local/go
+export PATH="${PATH}:${GOROOT}/bin"
+download_go
+export GOPATH="$(mktemp -d)"
+SRCDIR="$GOPATH/src/github.com/apache"
+mkdir -p "$SRCDIR"
+ln -s "$PWD" "$SRCDIR/trafficcontrol"
+
+cd "$SRCDIR/trafficcontrol/traffic_ops/traffic_ops_golang"
+
+/usr/local/go/bin/go get -v golang.org/x/net/publicsuffix\
+	golang.org/x/crypto/ed25519 \
+	golang.org/x/crypto/scrypt \
+	golang.org/x/net/idna \
+	golang.org/x/net/ipv4 \
+	golang.org/x/net/ipv6 \
+	golang.org/x/sys/unix \
+	golang.org/x/text/secure/bidirule > /dev/null
+/usr/local/go/bin/go build . > /dev/null
+
+openssl req -new -x509 -nodes -newkey rsa:4096 -out localhost.crt -keyout localhost.key -subj "/CN=tptests";
+
+resources="$(dirname "$0")"
+envsubst <"${resources}/cdn.json" >cdn.conf
+cp "${resources}/database.json" database.conf
+
+export $(<"${ciab_dir}/variables.env" sed '/^#/d') # defines TV_ADMIN_USER/PASSWORD
+envsubst <"${resources}/riak.json" >riak.conf
+
+truncate --size=0 warning.log error.log # Removes output from previous API versions and makes sure files exist
+./traffic_ops_golang --cfg ./cdn.conf --dbcfg ./database.conf -riakcfg riak.conf &
+tail -f warning.log 2>&1 | color_and_prefix "${yellow_bg}" 'Traffic Ops' &
+tail -f error.log 2>&1 | color_and_prefix "${red_bg}" 'Traffic Ops' &
+
+cd "../../traffic_portal"
+npm ci
+bower install
+grunt dist
+
+cp "${resources}/config.js" ./conf/
+touch tp.log access.log
+sudo forever --minUptime 5000 --spinSleepTime 2000 -l ./tp.log start server.js &
+
+fqdn="https://localhost:8443/"
+while ! curl -Lvsk "${fqdn}api/3.0/ping" >/dev/null 2>&1; do
+  echo "waiting for TP/TO server to start on '${fqdn}'"
+  sleep 10
+done
+
+
+cd "test/end_to_end"
+cp "${resources}/conf.json" .
+
+sudo protractor ./conf.js
+CODE=$?
+
+if [ $CODE -ne 0 ]; then
+	docker logs -f "$trafficvault" 2>&1 |
+		color_and_prefix "$gray_bg" 'Traffic Vault';
+  tail -f tp.log 2>&1 | color_and_prefix "${gray_bg}" 'Forever' &
+  tail -f access.log 2>&1 | color_and_prefix "${gray_bg}" 'Traffic Portal' &
```

Comment:
These lines should run in the foreground so the logs are guaranteed to be printed before the action exits.